libtas: 1.4.5 -> 1.4.6 + several fixes

MichailiK opened the pull request:
Supersedes / Resolves #325049

Bumps libtas to [v1.4.6](<https://github.com/clementgallet/libTAS/releases/tag/v1.4.6>) ([changelog](<https://github.com/clementgallet/libTAS/blob/v1.4.6/CHANGELOG.md#146---2024-07-05>)) with fixes to issues I've encountered throughout upgrading this package & using the package for developing a TAS for Deltarune.

CC: @skyrina

## Things done

- 510a921fa4cad23baf4b8e6174ffaf3013e70806 Bump libTAS to 1.4.6
- 510a921fa4cad23baf4b8e6174ffaf3013e70806 Bump Lua dependency to 5.4 [as suggested here](<https://github.com/NixOS/nixpkgs/pull/325049#pullrequestreview-2161844705>) as well as in the [README](<https://github.com/clementgallet/libTAS/tree/v1.4.6?tab=readme-ov-file#install>) of libtas.
- f76a7e7ed5287b152e6ee77bc4ab020c7cbfb28d Fix `undefined symbol: SDL_Log` errors by applying the patch of [this upstream commit](<https://github.com/clementgallet/libTAS/commit/779ff0fb0f3accfc62949680d85ecf96b28d18ef>) which is not part of a stable release yet.
- 6664ef393b5b513d03a157179127df06784e4f2f Allows libTAS to run on X11 again (fixes `Could not import symbol XIGetSelectedEvents` errors)
- 4c90a4483206cc09d5ea11eb44448689e9c9ab0f Fixes audio capture when encoding movies. Previously, videos would be silent because libTAS could not find `libswresample4`.
- e300b37aecd531173307a09787c33f416142fd56 Add (enabled by default) override option to force libTAS to run in X11. libTAS fails to capture game inputs under Wayland & upstream states X11 is supported only.

If desired, some parts of the PR can be dropped or be split up into separate PRs. 

## Checklist

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MichailiK committed:
libtas: 1.4.5 -> 1.4.6

Bumps libtas to 1.4.6. The new version depends now on Lua 5.4.

MichailiK committed:
libtas: Fix "undefined symbol: SDL_Log" errors

MichailiK committed:
libtas: Fix "Could not import symbol XIGetSelectedEvents" error on X11

MichailiK committed:
libtas: Fix movie encode failing to capture audio

libTAS requires libswresample to capture & encode audio of an application. Otherwise, the end result would be a silent recording.

MichailiK commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`
Commit: `4c90a4483206cc09d5ea11eb44448689e9c9ab0f`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>libtas</li>
  </ul>
</details>

skyrina commented:
Thank you!! I think it would be a good idea to force it to run in X11 by default, but maybe it should be an override option?
I believe I've seen a few packages force it in the past.

MichailiK committed:
libtas: Add withForceX11 option

libTAS needs to be run under X11 to capture inputs. Since libTAS is a QT app, it can launch in Wayland though. Thus, add an override option (enabled by default) which passes a `QT_QPA_PLATFORM=xcb` environment variable to force QT to run on X11.
